docs/modules/ROOT/pages/tutorials/getting-started/java-terminal.adoc

```diff
@@ -0,0 +1,310 @@
+        data.version = "1.0-java";
+        return data;
+    }
+
+    @GET
+    @Produces(MediaType.TEXT_HTML)
+    @Consumes(MediaType.TEXT_HTML)
+    public TemplateInstance hello() {
+        FortuneData data = makeFortuneData();
+        return Templates.index(data);
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public FortuneData helloJson() {
+        FortuneData data = makeFortuneData();
+        return data;
+    }
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    @Consumes(MediaType.TEXT_PLAIN)
+    public String helloText() {
+        FortuneData data = makeFortuneData();
+        return data.toString();
+    }
+}
+--
+
+=== Run the Container
+
+This step is optional; you can easily test the application locally with Docker or Podman:
+
+[source,shell]
+--
+$ docker run --rm --publish 8080:8080 registry.gitlab.com/vshn/applications/fortune-java:latest
+$ podman run --rm --publish 8080:8080 registry.gitlab.com/vshn/applications/fortune-java:latest
+--
+
+Now you can test the application with `curl`:
+
+[source,shell]
+--
+$ curl http://localhost:8080 --header "Accept: application/json"
+$ curl http://localhost:8080 --header "Accept: text/plain"
+--
+
+You can also open a browser and navigate to http://localhost:8080 to get your fortune cookie of the day.
+
+=== Dockerfile
+
+The application includes a https://gitlab.com/vshn/applications/fortune-java/-/blob/master/Dockerfile[Dockerfile] ready to build a container fully compatible with OpenShift. The snippet below shows the instructions used to build the actual running image:
+
+[source,dockerfile,indent=0]
+--
+# Stage 3: assemble runtime image
+FROM quay.io/quarkus/quarkus-distroless-image:1.0
+ENV PATH=/
+COPY --from=fortune /usr/local/games/fortune /fortune
+COPY --from=fortune /usr/local/bin/strfile /strfile
+COPY --from=fortune /usr/local/bin/unstr /unstr
+COPY --from=fortune /usr/local/bin/rot /rot
+COPY --from=fortune /usr/local/local/share/games/fortunes /usr/local/local/share/games/fortunes
+COPY --from=quarkus /code/target/*-runner /application
+
+EXPOSE 8080
+
+# <1>
+USER nonroot
+
+CMD ["./application", "-Dquarkus.http.host=0.0.0.0"]
+--
+<1> This explicitly prevents the container from running as root; this is a requirement of OpenShift, and a good practice for images in general.
+
+You can use the `Dockerfile` above to build your own copy of the container, which you can then push to the registry of your choice:
+
+[source,shell]
+--
+$ git clone https://gitlab.com/vshn/applications/fortune-java.git
+$ cd fortune-java
+$ docker build -t fortune-java .
+$ podman build -t fortune-java .
+--
+
+== Step 1: Create a Project
+
+Follow these steps to login to {product} on your terminal, create a project, and to deploy the application:
+
+. Login to the {product} console with your web browser.
+. Click on your user name on the top right and select "Copy login command"
```

Comment:
As we're in the console how-to here, I suggest pointing out this way for login:

```
oc login --server=https://api.cloudscale-lpg-2.appuio.cloud:6443
```

You then get the URL to directly access the token page. This way you're much faster than going through a full weconsole load cycle (See also [APUB-35](https://ticket.vshn.net/browse/APUB-35)).

Reply:
Done